merkletree/root.go

```diff
@@ -0,0 +1,78 @@
+package merkletree
+
+import (
+	"hash"
+)
+
+// TODO: Explain how and why this works here. Why + how should not be part of
+// the docstring of a function, that should only explain 'what'. Why and how
+// belong elsewhere.
```

Comment:
not necessarily true; see the docstring of `json.Marshal`, for example. I think an explanatory docstring would be fine. Preferably it would be on `Tree`.